Currently, patch-id trips over the diff extension for marking
the absence of newline at EOF.

Fix it. (Ignore it, it's whitespace.)

This uses the same detection rationale as in buitlin/apply.c, which was
introduced in

433ef8a ([PATCH] Make git-apply understand incomplete lines in non-C locales, 2005-09-04)

Signed-off-by: Michael J Gruber <git <at> drmicha.warpmail.net>
---
 builtin/patch-id.c  |    2 ++
 t/t4204-patch-id.sh |    2 +-
 2 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/builtin/patch-id.c b/builtin/patch-id.c
index 5125300..e1c3cb9 100644
--- a/builtin/patch-id.c
+++ b/builtin/patch-id.c
@@ -73,6 +73,8 @@ int get_one_patchid(unsigned char *next_sha1, git_SHA_CTX *ctx)
 			p += 7;
 		else if (!memcmp(line, "From ", 5))
 			p += 5;
+		else if (!memcmp(line, "\\ ", 2) && strlen(line)>=12)
+			continue;

 		if (!get_sha1_hex(p, next_sha1)) {
 			found_next = 1;

